File: python/Euler_021.py
def sum_proper_divisors(n):
    return sum(i for i in range(1, n) if n % i == 0)


def is_amicable(a):
    b = sum_proper_divisors(a)
    return b != a and sum_proper_divisors(b) == a


amicable_sum = sum(i for i in range(2, 10000) if is_amicable(i))
print(amicable_sum)

File: java/Euler_021.java

public class Euler_021 {

    public static int sumProperDivisors(int n) {
        int sumOfProperDivisors = 0;
        for (int i = 1; i < n; i++) {
            if (n % i == 0) {
                sumOfProperDivisors += i;
            }
        }
        return sumOfProperDivisors;
    }

    public static boolean isAmicable(int a) {
        int b = sumProperDivisors(a);
        return b != a && sumProperDivisors(b) == a;
    }

    public static void main(String[] args) {
        int sumAmicableNums = 0;
        for (int a = 2; a < 10000; a++) {
            if (isAmicable(a)) {
                sumAmicableNums += a;
            }
        }
        System.out.println(sumAmicableNums);
    }

}
